src/docdb/tree/DocDBDocumentTreeItem.ts

```diff
@@ -74,11 +79,15 @@ export class DocDBDocumentTreeItem implements IAzureTreeItem {
             throw new Error(`The "_self" and "_etag" fields are required to update a document`);
         }
         else {
+            let options = { accessCondition: { type: 'IfMatch', condition: newData._etag } };
+            if (this.partitionKeyValue) { //empty object is truth-y
```

Comment:
Is it necessary to avoid assigning the partition key value if it's falsy?  How you considered whether 0, empty string, null, etc. are valid partition keys?

Reply:
This is the error returned for a fixed size collection, if I pass 0, "" or {} as the partitionKey for a collection with no partition key (aka fixed size at this point): 
`Error: Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.`

Currently, this is how the API likes to receive partitionKeys: 
1. Key exists for collection, document has a value - give it the value, 
2. Key exists for collection, document has no value - give it an empty object, as outlined [in the node-documentdb workaround](https://github.com/Azure/azure-documentdb-node/issues/222)
3. No key exists (_currently_ corresponds to fixed size) do not give it a partition key value.

Reply:
Again, I'm not sure why the "if" is even necessary. If there's no partition key, partitionKeyValue will be undefined, and set as undefined into the options.  I wouldn't expect that to be a problem (although I have seen code where it is).

Reply:
I placed it there due a typescript compile issue (concerning the type of the partitionKey field). I'll move it back and <any> cast it. Seems better that way.